src/contractor/contractor.cpp

```diff
@@ -787,9 +787,17 @@ EdgeID Contractor::LoadEdgeExpandedGraph(
                     }
                     else
                     {
-                        // This edge is blocked, we don't need to continue updating
-                        new_weight = INVALID_EDGE_WEIGHT;
-                        break;
+                        // If we hit a 0-speed edge, then it's effectively not traversible.
+                        // We don't want to include it in the edge_based_edge_list, so
+                        // we jump to the end of the parent loop and skip the remaineder
+                        // of the logic.  This is effectively a `continue` for the outer loop.
+                        // This is one of the few good use-cases for `goto` (a break/continue
+                        // on a parent loop), consider carefully the extra complexity involved
+                        // if you decide to remove this.
+                        // Note, we need to increment the penaltyblock here because the goto
+                        // skips the regular incrementing lower down.
+                        penaltyblock++;
+                        goto skip_this_edge;
```

Comment:
No `goto` calls even if they look elegant. This creates precedence I don't want to see in the code base. should be easy to replace by a boolean and a break.